libtenzir/builtins/operators/tune.cpp

```diff
@@ -0,0 +1,182 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/tql2/plugin.hpp>
+
+namespace tenzir::plugins::tune {
+
+namespace {
+
+struct tune_args {
+  std::optional<located<duration>> idle_after = {};
+  std::optional<located<uint64_t>> min_demand_elements = {};
+  std::optional<located<uint64_t>> max_demand_elements = {};
+  std::optional<located<uint64_t>> max_demand_batches = {};
+  std::optional<located<bool>> detached = {};
+
+  friend auto inspect(auto& f, tune_args& x) {
+    return f.object(x).fields(
+      f.field("idle_after", x.idle_after),
+      f.field("min_demand_elements", x.min_demand_elements),
+      f.field("max_demand_elements", x.max_demand_elements),
+      f.field("max_demand_batches", x.max_demand_batches),
+      f.field("detached", x.detached));
+  }
+};
+
+class tune_operator final : public operator_base {
+public:
+  tune_operator() = default;
+
+  explicit tune_operator(operator_ptr op, tune_args args)
+    : op_{std::move(op)}, args_{std::move(args)} {
+    if (auto* op = dynamic_cast<tune_operator*>(op_.get())) {
+      op_ = std::move(op->op_);
+    }
```

Comment:
It seems this isn't quite right, you would always use only the outermost tune options.

Reply:
Good catch, this was incorrectly copied over from `local` and `remote`, where this flattening is possible.

Reply:
Fixed in b8fcc77a2be9dee673c74e87206f17a73f984cbf